src/test/java/org/cactoos/io/TeeInputPartATest.java

```diff
@@ -0,0 +1,979 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.io;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.function.Supplier;
+import org.cactoos.text.TextOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test case for {@link TeeInput}. Part A.
+ * @author Roman Proshin (roman@proshin.org)
+ * @version $Id$
+ * @since 1.0
+ * @checkstyle JavadocMethodCheck (1000 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (1000 lines)
+ */
+@SuppressWarnings(
+    {"PMD.TooManyMethods", "PMD.AvoidDuplicateLiterals", "PMD.GodClass"}
+)
+public final class TeeInputPartATest {
+
+    /**
+     * Test content for all the tests in this class.
+     */
+    private static final String CONTENT =
+        "Hello, товарищ äÄ üÜ öÖ and ß";
+
+    @Test
+    public void copiesFromUrlToPath() throws IOException {
```

Comment:
@proshin-roman We have to create multiple test classes, since there are so many tests. An idea would be to create test classes based on the source of the TeeInput. E.g. ``TeeInputFromUriTestCase``, ``TeeInputFromPathTestCase`` etc and each of these test files would contain tests for TeeInput from Uri to everything else, from Path to everything else.